drivers/flash/bch.c

```diff
@@ -0,0 +1,428 @@
+/*
+ * Copyright (c) 2022-2025 Macronix International Co., Ltd.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include "bch.h"
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#ifndef CONFIG_SPI_NAND_BCH_HEAP_SIZE
+#define CONFIG_SPI_NAND_BCH_HEAP_SIZE 51200
+#endif
+
+#define MAX_GEN_POLY_SIZE 169
+
+K_HEAP_DEFINE(bch_heap, CONFIG_SPI_NAND_BCH_HEAP_SIZE);
+LOG_MODULE_REGISTER(bch, CONFIG_FLASH_LOG_LEVEL);
+
+static void *bch_alloc(size_t size, int *err)
+{
+	void *ptr = NULL;
+
+	if (*err == 0) {
+		ptr = k_heap_alloc(&bch_heap, size, K_NO_WAIT);
+	}
+	if (ptr == NULL) {
+		*err = 1;
+	}
+	return ptr;
+}
+
+static inline uint32_t swap32_byte(uint32_t val)
+{
+	return (val & 0xff000000) >> 24 |
+			(val & 0x00ff0000) >> 8 |
+			(val & 0x0000ff00) << 8 |
+			(val & 0x000000ff) << 24;
+}
+
+static inline int mod(bch_t *bch, uint32_t v)
+{
+	while (v >= bch->n) {
+		v -= bch->n;
+		v = (v & bch->n) + (v >> bch->m);
+	}
+	return v;
+}
+
+static void build_syndrome(bch_t *bch)
+{
+	int i, j;
+	int ecc_bits;
+	uint32_t *ecc;
+
+	memset(bch->syn, 0, 2*bch->t*sizeof(*bch->syn));
```

Comment:
Weird that checkpatch doesn't complain here.

```suggestion
	memset(bch->syn, 0, 2 * bch->t * sizeof(*bch->syn));
```

Reply:
Resolved